application/src/test/java/org/thingsboard/server/service/ruleengine/DefaultRuleEngineCallServiceTest.java

```diff
@@ -0,0 +1,159 @@
+        ruleEngineCallService = new DefaultRuleEngineCallService(tbClusterServiceMock);
+        ReflectionTestUtils.setField(ruleEngineCallService, "executor", executor);
+        ReflectionTestUtils.setField(ruleEngineCallService, "requests", requests);
+    }
+
+    @AfterEach
+    void tearDown() {
+        requests.clear();
+        if (executor != null) {
+            executor.shutdownNow();
+        }
+    }
+
+    @Test
+    void givenRequest_whenProcessRestApiCallToRuleEngine_thenPushMsgToRuleEngine() {
+        long timeout = 100L;
+        long expTime = System.currentTimeMillis() + timeout;
+        HashMap<String, String> metaData = new HashMap<>();
+        UUID requestId = UUID.randomUUID();
+        metaData.put("serviceId", "core");
+        metaData.put("requestUUID", requestId.toString());
+        metaData.put("expirationTime", Long.toString(expTime));
+        TbMsg msg = TbMsg.newMsg(DataConstants.MAIN_QUEUE_NAME, TbMsgType.REST_API_REQUEST, TENANT_ID, new TbMsgMetaData(metaData), "{\"key\":\"value\"}");
+        Consumer<TbMsg> anyConsumer = TbMsg::getData;
+        ruleEngineCallService.processRestApiCallToRuleEngine(TENANT_ID, requestId, msg, true, anyConsumer);
+
+        assertThat(requests.size()).isEqualTo(1);
+        assertThat(requests.get(requestId)).isEqualTo(anyConsumer);
+        verify(tbClusterServiceMock).pushMsgToRuleEngine(TENANT_ID, TENANT_ID, msg, true, null);
+    }
+
+    @Test
+    void givenSmallTimeout_whenProcessRestApiCallToRuleEngine_thenDoesNotReturnResponse() {
+        long timeout = 1L;
+        long expTime = System.currentTimeMillis() + timeout;
+        HashMap<String, String> metaData = new HashMap<>();
+        UUID requestId = UUID.randomUUID();
+        metaData.put("serviceId", "core");
+        metaData.put("requestUUID", requestId.toString());
+        metaData.put("expirationTime", Long.toString(expTime));
+        TbMsg msg = TbMsg.newMsg(DataConstants.MAIN_QUEUE_NAME, TbMsgType.REST_API_REQUEST, TENANT_ID, new TbMsgMetaData(metaData), "{\"key\":\"value\"}");
+
+        Consumer<TbMsg> anyConsumer = TbMsg::getData;
+        doAnswer(invocation -> {
+            //check the presence of request in the map after pushMsgToRuleEngine()
+            assertThat(requests.size()).isEqualTo(1);
+            assertThat(requests.get(requestId)).isEqualTo(anyConsumer);
+            return null;
+        }).when(tbClusterServiceMock).pushMsgToRuleEngine(any(), any(), any(), anyBoolean(), any());
+        ruleEngineCallService.processRestApiCallToRuleEngine(TENANT_ID, requestId, msg, true, anyConsumer);
+
+        verify(tbClusterServiceMock).pushMsgToRuleEngine(TENANT_ID, TENANT_ID, msg, true, null);
+        //check map is empty after scheduleTimeout()
+        Awaitility.await("Await until request was deleted from map due to timeout")
+                .pollDelay(25, TimeUnit.MILLISECONDS)
+                .atMost(10, TimeUnit.SECONDS)
+                .until(requests::isEmpty);
+    }
```

Comment:
let's leave only the second test since they both do almost the same check. Please rename it to something like:

givenRequest_whenProcessRestApiCallToRuleEngine_thenPushMsgToRuleEngineAndRemoveDueTimeout